addons/payment/static/src/js/payment_form.js

```diff
@@ -0,0 +1,250 @@
+odoo.define('payment.payment_form', function (require){
+    "use strict";
+
+    var ajax = require('web.ajax');
+    var core = require('web.core');
+    var Dialog = require("web.Dialog");
+    var Widget = require("web.Widget");
+
+
+    var PaymentForm = Widget.extend({
+        events: {
+            'click #o_payment_form_pay': function(ev) {
+                ev.stopPropagation();
+                ev.preventDefault();
+                var form = this.el;
+                var checked_radio = this.$('input[type="radio"]:checked');
+                var self = this;
+                var button = ev.target;
+
+                // first we check that the user has selected a payment method
+                if(checked_radio.length == 1) {
+                    checked_radio = checked_radio[0];
+                    // if the user is adding a new payment
+                    if(this.isNewPaymentRadio(checked_radio)) {
+                        var acquirer_id = this.getAcquirerIdFromRadio(checked_radio);
+                        var acquirer_form = this.$('#o_payment_add_token_acq_' + acquirer_id);
+                        // we retrieve all the input inside the acquirer form and 'serialize' them to an indexed array
+                        var form_data = this.getFormData($('input', acquirer_form));
+                        var ds = $('input[name="data_set"]', acquirer_form)[0];
+                        $(button).attr('disabled', true);
+                        $(button).prepend('<i class="fa fa-refresh fa-spin"></i>');
```

Comment:
you probably need a space after the icon otherwise it looks bad
```javascript
$(button).prepend('<span class="o_loader"><i class="fa fa-refresh fa-spin"></i>&nbsp;</span>');
```

and
```javascript
$(button).find('span.o_loader').remove();
```